### PR TITLE
[staging/azure] azure_utils: fix range iterator issue in convertMaptoMapPointer

### PR DESCRIPTION
**Pull request by gaurav1086**
Signed-off-by: Gaurav Singh <gaurav1086@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Go uses the same address variable while iterating in a range, so use a copy when using its address.
**Which issue(s) this PR fixes**:

Fixes #
N/A
**Special notes for your reviewer**:
N/A
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
/ok-to-test
/priority important-soon
/sig cloud-provider
/area provider/azure
@gaurav1086 thanks for the contribution.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func ConvertTagsToMap(tags string) (map[string]string, error) {
 func convertMaptoMapPointer(origin map[string]string) map[string]*string {
 	newly := make(map[string]*string)
 	for k, v := range origin {
-		newly[k] = &v
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go:113]: @ZeroMagic could you help review this? Thanks.

Comment (andyzhangx) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go:113]: looks like it's a common mistake in golang coding, I remember that, thanks.

Comment (andyzhangx) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_utils.go:113]: there is compiling error, try this:
```
        for k, v := range origin {
                value := v
                newly[k] = &value
        }
```

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by andyzhangx** [comment]
there is a PR to fix the windows test failure: https://github.com/kubernetes/kubernetes/pull/93063

**Review by ZeroMagic — APPROVED** [review]
/milestone v1.19
/lgtm
/approve

**Comment by ZeroMagic** [comment]
Thanks for the fix.

**Review by feiskyer — APPROVED** [review]
/milestone v1.19

**Pull request merged**